# This script takes the filtered vcf files from various experiments and converts them into flat files 
# that contian mean genotype probabilities. Each column represents a sample and each row is a locus 
setwd("/media/raglandlab/ExtraDrive4/Clines_3/")
library(vcfR) 
#All of the SQL scripts will not work unless connected to our local server. The data folder of the repo will contain a txt file of the data 
#that was generated by these commands that can be used to reproduce the analysis here. 
library(RMySQL)

source("./src/clines3_functions.R")

#Load in the data for the set of shared SNPs across all data sets, this set of loci is derived from Doellman et al. 2019. 
#This set of loci underwent a series of filters, such as HWE to look for null alleles. 

jeff_snps <- read.table("./data/position_fix.txt")
jeff_snps <- as.vector(jeff_snps$V1)

#Data was polarized such that the reference allele was the allele that was most common in the hawthorn population at 
#Grant, MI. All vcf files in this analysis were aligned to a consensus 'RADome' so all have the same reference and alternate 
#alleles.
#The script below creates a T/F list to indentify which loci needed to be switched or polarized to 
Das <- read.vcfR("./data/snps.ScottSelection.05-400-20.vcf", nrows = -1, skip = 0, cols = NULL, convertNA = TRUE, verbose = T)
PLs_grant <- extract.gt(Das, element = "PL", mask = F, as.numeric = F, return.alleles = F, IDtoRowNames = TRUE, extract = T)
grant_ids <- read.table("./data/SelectionIdDescriptors.txt")
ind <- grep("Haw_[a-z]_7d", grant_ids$V2)
grant_haw_ids <- grant_ids[ind,]
ind <- grep("Apple_[a-z]_7d", grant_ids$V2)
grant_apple_ids <- grant_ids[ind,]
ind <- colnames(PLs_grant) %in% grant_haw_ids$V1
PLs_grant_haw <- PLs_grant[,ind]
ind <- colnames(PLs_grant) %in% grant_apple_ids$V1
PLs_grant_apple <- PLs_grant[,ind]
ind <- rownames(PLs_grant) %in% jeff_snps
PLs_grant_haw <- PLs_grant_haw[ind,]
PLs_grant_apple <- PLs_grant_apple[ind,]
r <- nrow(PLs_grant_haw)
c <- ncol(PLs_grant_haw)
mat_grant_haw <- matrix(lapply(PLs_grant_haw, splitPls), nrow=r, ncol=c)
mat_grant_haw <- apply(mat_grant_haw,2,as.numeric)
r <- nrow(PLs_grant_apple)
c <- ncol(PLs_grant_apple)
mat_grant_apple <- matrix(lapply(PLs_grant_apple, splitPls), nrow=r, ncol=c)
mat_grant_apple <- apply(mat_grant_apple,2,as.numeric)
colnames(mat_grant_haw)<-colnames(PLs_grant_haw)
rownames(mat_grant_haw)<-rownames(PLs_grant_haw)
colnames(mat_grant_apple) <- colnames(PLs_grant_apple)
rownames(mat_grant_apple) <- rownames(PLs_grant_apple)
geno_freqs_haw <- matrix(apply(mat_grant_haw,1,freqEst), nrow = nrow(PLs_grant_haw), ncol = 1)
geno_freqs_apple <- matrix(apply(mat_grant_apple,1,freqEst), nrow = nrow(PLs_grant_haw), ncol = 1)
comp_mat <- cbind(geno_freqs_haw, geno_freqs_apple)
comp_mat_2 <- comp_mat[,1] - comp_mat[,2]
pol_ind <- vector(length = length(geno_freqs_haw))
#produce a logical vector of which genos need to be swapped 
for(i in 1:length(comp_mat_2)){
  if(comp_mat_2[i] > 0){
    pol_ind[i] <- TRUE
  }else{
    pol_ind[i] <- FALSE
  }
}

#Save the list so that it can be used in future scripts. 
save(pol_ind, file="./data/polarized_loci.Rdata")

###Then we'll also produce an index of all of the loci in our dataset, that were mapped in Ragland et al. 2017 Mol. Ecol. 

mydb = dbConnect(MySQL(), user='raglandlab', password='pomonella', dbname='PomUrbanaPoolseqSNP')

#Need to split jeff snps to query the database 
shared_snps <- matrix(unlist(strsplit(jeff_snps, "_")), ncol = 2, byrow = T)

for (i in 1:nrow(shared_snps)){
  dat<-dbGetQuery(mydb, paste("select * from RAD_linkage_matchPool where RAD_contig = '",shared_snps[i,1],"'and RAD_position = '",shared_snps[i,2],"'",sep=""))
  #dat<-dbGetQuery(mydb, paste("select * from annotation where snpId = '",dat$snpId,"'",sep=""))
  if (i==1) {scafs<-dat} else {scafs=rbind(scafs,dat)}
}

scafs$RAD_contig <- as.character(scafs$RAD_contig)
RAD_chr <- matrix(nrow=nrow(scafs), ncol=1)
for(i in 1:nrow(scafs)){
  a <- paste0(scafs[i,4:5], collapse="_")
  RAD_chr[i,1] <- a
}

#remove the mapped position on the Rhagoletis zepheryia genome - this is information on an unpublished genome, please contact the 
#authors if you would like access to it. no
scafs <- scafs[,-c(2,3,9)]

#use the file using the load command to get the mapping data 
#load("./data/Mapped_RAD_loci.Rdata")
### We can then make a bunch of smaller lists of loci for each chromosome and LD group - This is tedious, but it makes 
#generating the mean genotype files below more transparent and easier to undertsand. 
#Remember there are 5 chromosomes and loci that were mapped were assigned to Low, Intermediate, and High LD groups, see 
#manuscript text and Rangland et al. 2017 Mol. Ecol. for more details. 

#The commented rm script below was run to produce the RData file below that contains positional information for each chromosome 
#LD group.
#rm(list=setdiff(ls(), c("mat_grant_haw", "scafs", "RAD_chr"))) 
#ls() 

#Total chromsomes 
chr1_ind <- scafs$chromosome=="1"
chr1_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr1_ind]
chr2_ind <- scafs$chromosome=="2"
chr2_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_ind]
chr3_ind <- scafs$chromosome=="3"
chr3_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr3_ind]
chr4_ind <- scafs$chromosome=="4"
chr4_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr4_ind]
chr5_ind <- scafs$chromosome=="5"
chr5_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr5_ind]

#All chromosomes
all_chr_ind_p <- chr1_ind_p | chr2_ind_p | chr3_ind_p | chr4_ind_p | chr5_ind_p

#Chromosome 1 LD groups 
chr1_L_ind <- scafs$chromosome=="1" & scafs$LDgr =="L"
chr1_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr1_L_ind]
chr1_M_ind <- scafs$chromosome=="1" & scafs$LDgr =="M"
chr1_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr1_M_ind]
chr1_H_ind <- scafs$chromosome=="1" & scafs$LDgr =="H"
chr1_H_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr1_H_ind]

#Chromosome 2 LD groups 
chr2_L_ind <- scafs$chromosome=="2" & scafs$LDgr =="L"
chr2_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_L_ind]
chr2_M_ind <- scafs$chromosome=="2" & scafs$LDgr =="M"
chr2_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_M_ind]
chr2_L_ind <- scafs$chromosome=="2" & scafs$LDgr =="L"
chr2_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_L_ind]
chr2_M_ind <- scafs$chromosome=="2" & scafs$LDgr =="M"
chr2_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_M_ind]
chr2_H1_ind <- scafs$chromosome=="2" & scafs$LDgr =="H1"
chr2_H2_ind <- scafs$chromosome=="2" & scafs$LDgr =="H2"
chr2_H3_ind <- scafs$chromosome=="2" & scafs$LDgr =="H3"
chr2_H4_ind <- scafs$chromosome=="2" & scafs$LDgr =="H4"
chr2_H5_ind <- scafs$chromosome=="2" & scafs$LDgr =="H5"
chr2_H6_ind <- scafs$chromosome=="2" & scafs$LDgr =="H6"
chr2_H7_ind <- scafs$chromosome=="2" & scafs$LDgr =="H7"
chr2_H8_ind <- scafs$chromosome=="2" & scafs$LDgr =="H8"
chr2_H_ind <- chr2_H1_ind | chr2_H2_ind | chr2_H3_ind | chr2_H4_ind | chr2_H5_ind | chr2_H6_ind | chr2_H7_ind | chr2_H8_ind
chr2_H_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr2_H_ind]

#Chromosome 3 LD groups
chr3_L_ind <- scafs$chromosome=="3" & scafs$LDgr =="L"
chr3_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr3_L_ind]
chr3_M_ind <- scafs$chromosome=="3" & scafs$LDgr =="M"
chr3_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr3_M_ind]
chr3_H_ind <- scafs$chromosome=="3" & scafs$LDgr =="H"
chr3_H_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr3_H_ind]

chr4_L_ind <- scafs$chromosome=="4" & scafs$LDgr =="L"
chr4_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr4_L_ind]
chr4_M_ind <- scafs$chromosome=="4" & scafs$LDgr =="M"
chr4_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr4_M_ind]
chr4_H_ind <- scafs$chromosome=="4" & scafs$LDgr =="H"
chr4_H_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr4_H_ind]

chr5_L_ind <- scafs$chromosome=="5" & scafs$LDgr =="L"
chr5_L_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr5_L_ind]
chr5_M_ind <- scafs$chromosome=="5" & scafs$LDgr =="M"
chr5_M_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr5_M_ind]
chr5_H_ind <- scafs$chromosome=="5" & scafs$LDgr =="H"
chr5_H_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[chr5_H_ind]

all_chr_ind <- scafs$chromosome=="1" | scafs$chromosome=="2" | scafs$chromosome=="3" | scafs$chromosome=="4" | scafs$chromosome=="5"
all_LDL <- scafs$LDgr == "L"
all_LDM <- scafs$LDgr == "M"
all_LDH <- scafs$LDgr == "H" | scafs$LDgr == "H1" | scafs$LDgr == "H2" | scafs$LDgr == "H3" | scafs$LDgr == "H4" | scafs$LDgr == "H5" | scafs$LDgr == "H6" | scafs$LDgr == "H7" | scafs$LDgr == "H8"
all_chr_ind_p <- rownames(mat_grant_haw) %in% RAD_chr[all_chr_ind]
all_LDL_p <- rownames(mat_grant_haw) %in% RAD_chr[all_LDL]
all_LDM_p <- rownames(mat_grant_haw) %in% RAD_chr[all_LDM]
all_LDH_p <- rownames(mat_grant_haw) %in% RAD_chr[all_LDH]

#Analayses involving the diapaus inensity data will use only female individuals for any comparisons 
# involving chromosome 5. See main text for details, but briefly chromosome 5 harbors the sex determining 
# region so to correct for any sex based variation on chromoome 5 we chose to only use females. 

chr1_4_all_P <- (scafs$chromosome=="1" | scafs$chromosome=="2" | scafs$chromosome=="3" | scafs$chromosome=="4")

chr1_4_LDL_p <- scafs$LDgr == "L" & (scafs$chromosome=="1" | scafs$chromosome=="2" | scafs$chromosome=="3" | scafs$chromosome=="4")
chr1_4_LDM_p <- scafs$LDgr == "M" & (scafs$chromosome=="1" | scafs$chromosome=="2" | scafs$chromosome=="3" | scafs$chromosome=="4")
chr1_4_LDH_p <- scafs$LDgr == "H" & (scafs$chromosome=="1" | scafs$chromosome=="2" | scafs$chromosome=="3" | scafs$chromosome=="4")

save.image(file="./data/Mapped_RAD_loci.Rdata")

#Geographic survey data, this data was originally published in Doelman et al. 2019 Ecol & Evo

#Read in the vcf file for the geogrphic data 
Das <- read.vcfR("./data/clinal.UG.GATK.all.good.filtered.vcf", nrows = -1, skip = 0, cols = NULL, convertNA = TRUE, verbose = T)
PLs <- extract.gt(Das, element = "PL", mask = F, as.numeric = F, return.alleles = F, IDtoRowNames = TRUE, extract = T)
#vcfR objects take a lot of memory so we'll delete them whenever we can. 
rm(Das)
r <- nrow(PLs)
c <- ncol(PLs)
mat <- matrix(lapply(PLs, splitPls), nrow=r, ncol=c)
rownames(mat)<-rownames(PLs)
colnames(mat)<-colnames(PLs)
#Index the shared loci
ind <- rownames(mat) %in% jeff_snps 
#Subset the mean genotypes for those shared loci
mat <- mat[ind,]
rnames <- rownames(mat)
mat<- apply(mat,2,as.numeric)
row.names(mat) <- rnames
rm(PLs)

#Polarize the data
geo_scored <- matrix(nrow = nrow(mat), ncol = ncol(mat))
for(i in 1:length(pol_ind)){
  for(j in 1:ncol(geo_scored))
    if(pol_ind[i]==FALSE){
      geo_scored[i,j] <- mat[i,j]
    }else if(pol_ind[i]==TRUE){
      geo_scored[i,j] <- (2-mat[i,j])
    }
}
colnames(geo_scored) <- colnames(mat)
rownames(geo_scored) <- rownames(mat)

#split up by host race and by location.
IDs <- read.table("./data/clinal.pops.txt", header=F)
Brazos_haw_Ids <- as.character(IDs$V1[which(IDs$V4 == "BrazosBendHaw")])
Brazos_haw_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Brazos_haw_Ids)]
Fen_haw_Ids <- as.character(IDs$V1[which(IDs$V4 == "FennvilleHaw")]) 
Fen_haw_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Fen_haw_Ids)]
Fen_apple_Ids <- as.character(IDs$V1[which(IDs$V4 == "FennvilleApple")])
Fen_apple_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Fen_apple_Ids)]
Dow_haw_Ids <- as.character(IDs$V1[which(IDs$V4 == "DowagiacHaw")]) 
Dow_haw_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Dow_haw_Ids)]
Dow_apple_Ids <- as.character(IDs$V1[which(IDs$V4 == "DowagiacApple")])
Dow_apple_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Dow_apple_Ids)]
Urb_haw_Ids <- as.character(IDs$V1[which(IDs$V4 == "UrbanaHaw")]) 
Urb_haw_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Urb_haw_Ids)]
Urb_apple_Ids <- as.character(IDs$V1[which(IDs$V4 == "UrbanaApple")])
Urb_apple_geo_scored <- geo_scored[,which(colnames(geo_scored) %in% Urb_apple_Ids)]

#Ok, Now we can write the data tables by chromosome, LD group, and both
#write.table(Brazos_haw_geo_scored, file = "BrazosHaw_genos_AllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#set the working director to be in the data directory, 
setwd("./data/")
write.table(Fen_haw_geo_scored, file="FenHaw_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored, file="FenApple_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored, file="DowHaw_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored, file="DowApple_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored, file="UrbHaw_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored, file="UrbApple_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

##All mapped loci 
#write.table(Brazos_haw_geo_scored[all_chr_ind_p,], file = "BrazosHaw_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[all_chr_ind_p,], file="FenHaw_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[all_chr_ind_p,], file="FenApple_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[all_chr_ind_p,], file="DowHaw_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[all_chr_ind_p,], file="DowApple_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[all_chr_ind_p,], file="UrbHaw_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[all_chr_ind_p,], file="UrbApple_genos_AllMap_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LDL
#write.table(Brazos_haw_geo_scored[all_LDL_p,], file = "BrazosHaw_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[all_LDL_p,], file="FenHaw_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[all_LDL_p,], file="FenApple_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[all_LDL_p,], file="DowHaw_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[all_LDL_p,], file="DowApple_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[all_LDL_p,], file="UrbHaw_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[all_LDL_p,], file="UrbApple_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LDM
#write.table(Brazos_haw_geo_scored[all_LDM_p,], file = "BrazosHaw_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[all_LDM_p,], file="FenHaw_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[all_LDM_p,], file="FenApple_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[all_LDM_p,], file="DowHaw_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[all_LDM_p,], file="DowApple_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[all_LDM_p,], file="UrbHaw_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[all_LDM_p,], file="UrbApple_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LDH
#write.table(Brazos_haw_geo_scored[all_LDH_p,], file = "BrazosHaw_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[all_LDH_p,], file="FenHaw_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[all_LDH_p,], file="FenApple_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[all_LDH_p,], file="DowHaw_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[all_LDH_p,], file="DowApple_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[all_LDH_p,], file="UrbHaw_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[all_LDH_p,], file="UrbApple_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

##Fenville, MI all chromosomes and LD groups
#haw
write.table(Fen_haw_geo_scored[chr1_ind_p,], file= "FenHaw_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr1_L_ind_p,], file= "FenHaw_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr1_M_ind_p,], file= "FenHaw_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr1_H_ind_p,], file= "FenHaw_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr2_ind_p,], file= "FenHaw_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr2_L_ind_p,], file= "FenHaw_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr2_M_ind_p,], file= "FenHaw_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr2_H_ind_p,], file= "FenHaw_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr3_ind_p,], file= "FenHaw_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr3_L_ind_p,], file= "FenHaw_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr3_M_ind_p,], file= "FenHaw_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr3_H_ind_p,], file= "FenHaw_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr4_ind_p,], file= "FenHaw_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr4_L_ind_p,], file= "FenHaw_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr4_M_ind_p,], file= "FenHaw_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr4_H_ind_p,], file= "FenHaw_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr5_ind_p,], file= "FenHaw_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr5_L_ind_p,], file= "FenHaw_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr5_M_ind_p,], file= "FenHaw_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_haw_geo_scored[chr5_H_ind_p,], file= "FenHaw_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#apple
write.table(Fen_apple_geo_scored[chr1_ind_p,], file= "FenApple_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr1_L_ind_p,], file= "FenApple_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr1_M_ind_p,], file= "FenApple_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr1_H_ind_p,], file= "FenApple_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr2_ind_p,], file= "FenApple_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr2_L_ind_p,], file= "FenApple_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr2_M_ind_p,], file= "FenApple_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr2_H_ind_p,], file= "FenApple_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr3_ind_p,], file= "FenApple_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr3_L_ind_p,], file= "FenApple_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr3_M_ind_p,], file= "FenApple_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr3_H_ind_p,], file= "FenApple_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr4_ind_p,], file= "FenApple_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr4_L_ind_p,], file= "FenApple_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr4_M_ind_p,], file= "FenApple_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr4_H_ind_p,], file= "FenApple_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr5_ind_p,], file= "FenApple_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr5_L_ind_p,], file= "FenApple_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr5_M_ind_p,], file= "FenApple_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Fen_apple_geo_scored[chr5_H_ind_p,], file= "FenApple_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

# Dowagiac, MI 
#haw
write.table(Dow_haw_geo_scored[chr1_ind_p,], file= "DowHaw_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr1_L_ind_p,], file= "DowHaw_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr1_M_ind_p,], file= "DowHaw_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr1_H_ind_p,], file= "DowHaw_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr2_ind_p,], file= "DowHaw_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr2_L_ind_p,], file= "DowHaw_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr2_M_ind_p,], file= "DowHaw_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr2_H_ind_p,], file= "DowHaw_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr3_ind_p,], file= "DowHaw_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr3_L_ind_p,], file= "DowHaw_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr3_M_ind_p,], file= "DowHaw_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr3_H_ind_p,], file= "DowHaw_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr4_ind_p,], file= "DowHaw_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr4_L_ind_p,], file= "DowHaw_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr4_M_ind_p,], file= "DowHaw_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr4_H_ind_p,], file= "DowHaw_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr5_ind_p,], file= "DowHaw_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr5_L_ind_p,], file= "DowHaw_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr5_M_ind_p,], file= "DowHaw_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_haw_geo_scored[chr5_H_ind_p,], file= "DowHaw_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#apple
write.table(Dow_apple_geo_scored[chr1_ind_p,], file= "DowApple_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr1_L_ind_p,], file= "DowApple_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr1_M_ind_p,], file= "DowApple_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr1_H_ind_p,], file= "DowApple_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr2_ind_p,], file= "DowApple_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr2_L_ind_p,], file= "DowApple_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr2_M_ind_p,], file= "DowApple_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr2_H_ind_p,], file= "DowApple_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr3_ind_p,], file= "DowApple_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr3_L_ind_p,], file= "DowApple_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr3_M_ind_p,], file= "DowApple_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr3_H_ind_p,], file= "DowApple_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr4_ind_p,], file= "DowApple_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr4_L_ind_p,], file= "DowApple_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr4_M_ind_p,], file= "DowApple_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr4_H_ind_p,], file= "DowApple_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr5_ind_p,], file= "DowApple_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr5_L_ind_p,], file= "DowApple_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr5_M_ind_p,], file= "DowApple_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Dow_apple_geo_scored[chr5_H_ind_p,], file= "DowApple_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

##Urbana, IL
#Haw
write.table(Urb_haw_geo_scored[chr1_ind_p,], file= "UrbHaw_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr1_L_ind_p,], file= "UrbHaw_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr1_M_ind_p,], file= "UrbHaw_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr1_H_ind_p,], file= "UrbHaw_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr2_ind_p,], file= "UrbHaw_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr2_L_ind_p,], file= "UrbHaw_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr2_M_ind_p,], file= "UrbHaw_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr2_H_ind_p,], file= "UrbHaw_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr3_ind_p,], file= "UrbHaw_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr3_L_ind_p,], file= "UrbHaw_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr3_M_ind_p,], file= "UrbHaw_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr3_H_ind_p,], file= "UrbHaw_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr4_ind_p,], file= "UrbHaw_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr4_L_ind_p,], file= "UrbHaw_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr4_M_ind_p,], file= "UrbHaw_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr4_H_ind_p,], file= "UrbHaw_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr5_ind_p,], file= "UrbHaw_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr5_L_ind_p,], file= "UrbHaw_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr5_M_ind_p,], file= "UrbHaw_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_haw_geo_scored[chr5_H_ind_p,], file= "UrbHaw_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#apple
write.table(Urb_apple_geo_scored[chr1_ind_p,], file= "UrbApple_genosChr1_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr1_L_ind_p,], file= "UrbApple_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr1_M_ind_p,], file= "UrbApple_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr1_H_ind_p,], file= "UrbApple_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr2_ind_p,], file= "UrbApple_genosChr2_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr2_L_ind_p,], file= "UrbApple_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr2_M_ind_p,], file= "UrbApple_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr2_H_ind_p,], file= "UrbApple_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr3_ind_p,], file= "UrbApple_genosChr3_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr3_L_ind_p,], file= "UrbApple_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr3_M_ind_p,], file= "UrbApple_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr3_H_ind_p,], file= "UrbApple_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr4_ind_p,], file= "UrbApple_genosChr4_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr4_L_ind_p,], file= "UrbApple_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr4_M_ind_p,], file= "UrbApple_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr4_H_ind_p,], file= "UrbApple_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr5_ind_p,], file= "UrbApple_genosChr5_all_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr5_L_ind_p,], file= "UrbApple_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr5_M_ind_p,], file= "UrbApple_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Urb_apple_geo_scored[chr5_H_ind_p,], file= "UrbApple_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Set the working directory back
setwd("../")
####Diapause termination data, these data were originally published in Ragland et al. 2017 Mol. Ecol.

Das <- read.vcfR("./data/PJM123.UG.GATK.all.filtered.vcf", nrows = -1, skip = 0, cols = NULL, convertNA = TRUE, verbose = T)
PLs <- extract.gt(Das, element = "PL", mask = F, as.numeric = F, return.alleles = F, IDtoRowNames = TRUE, extract = T)
rm(Das)
r <- nrow(PLs)
c <- ncol(PLs)
mat <- matrix(lapply(PLs, splitPls), nrow=r, ncol=c)
rownames(mat)<-rownames(PLs)
colnames(mat)<-colnames(PLs)
ind <- rownames(mat) %in% jeff_snps #this is where Jeff's method falls apart a bit 2 loci arent on the list
mat <- mat[ind,]
rnames <- rownames(mat)
mat<- apply(mat,2,as.numeric)
row.names(mat) <- rnames
ecl_scored <- matrix(nrow = nrow(mat), ncol = ncol(mat))
for(i in 1:length(pol_ind)){
  for(j in 1:ncol(ecl_scored))
    if(pol_ind[i]==FALSE){
      ecl_scored[i,j] <- mat[i,j]
    }else if(pol_ind[i]==TRUE){
      ecl_scored[i,j] <- (2-mat[i,j])
    }
}
colnames(ecl_scored) <- colnames(mat)
rownames(ecl_scored) <- rownames(mat)

Ids <- read.table("./data/appleHawIds.txt", header=F) #rows 1-51 haw late, 52-101 haw early, 102 - 151 apple late 152 - 200 apple early 
Apple_early_mat <- ecl_scored[,which(colnames(ecl_scored) %in% Ids$V1[152:200])]
Apple_late_mat <- ecl_scored[,which(colnames(ecl_scored) %in% Ids$V1[102:151])]
Haw_early_mat <- ecl_scored[,which(colnames(ecl_scored) %in% Ids$V1[52:101])]
Haw_late_mat <- ecl_scored[,which(colnames(ecl_scored) %in% Ids$V1[1:51])]

setwd("./data/")
#Write the genotype files: 
#all genotypes
write.table(Apple_early_mat, file= "appleEarly_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat, file= "appleLate_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat, file= "hawEarly_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat, file= "hawLate_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Apple Early termination
#All chromosomes
write.table(Apple_early_mat[chr1_ind_p,], file= "appleEarly_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr2_ind_p,], file= "appleEarly_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr3_ind_p,], file= "appleEarly_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr4_ind_p,], file= "appleEarly_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr5_ind_p,], file= "appleEarly_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[all_chr_ind_p,], file= "appleEarly_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LD groups
write.table(Apple_early_mat[all_LDL_p,], file= "appleEarly_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[all_LDM_p,], file= "appleEarly_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[all_LDH_p,], file= "appleEarly_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(Apple_early_mat[chr1_L_ind_p,], file= "appleEarly_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr1_M_ind_p,], file= "appleEarly_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr1_H_ind_p,], file= "appleEarly_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(Apple_early_mat[chr2_L_ind_p,], file= "appleEarly_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr2_M_ind_p,], file= "appleEarly_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr2_H_ind_p,], file= "appleEarly_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(Apple_early_mat[chr3_L_ind_p,], file= "appleEarly_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr3_M_ind_p,], file= "appleEarly_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr3_H_ind_p,], file= "appleEarly_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(Apple_early_mat[chr4_L_ind_p,], file= "appleEarly_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr4_M_ind_p,], file= "appleEarly_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr4_H_ind_p,], file= "appleEarly_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(Apple_early_mat[chr5_L_ind_p,], file= "appleEarly_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr5_M_ind_p,], file= "appleEarly_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_early_mat[chr5_H_ind_p,], file= "appleEarly_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Apple late termination 
#All chromosomes
write.table(Apple_late_mat[chr1_ind_p,], file= "appleLate_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr2_ind_p,], file= "appleLate_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr3_ind_p,], file= "appleLate_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr4_ind_p,], file= "appleLate_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr5_ind_p,], file= "appleLate_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[all_chr_ind_p,], file= "appleLate_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LD groups
write.table(Apple_late_mat[all_LDL_p,], file= "appleLate_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[all_LDM_p,], file= "appleLate_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[all_LDH_p,], file= "appleLate_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")


#Chromosome 1 LD
write.table(Apple_late_mat[chr1_L_ind_p,], file= "appleLate_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr1_M_ind_p,], file= "appleLate_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr1_H_ind_p,], file= "appleLate_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(Apple_late_mat[chr2_L_ind_p,], file= "appleLate_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr2_M_ind_p,], file= "appleLate_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr2_H_ind_p,], file= "appleLate_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(Apple_late_mat[chr3_L_ind_p,], file= "appleLate_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr3_M_ind_p,], file= "appleLate_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr3_H_ind_p,], file= "appleLate_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(Apple_late_mat[chr4_L_ind_p,], file= "appleLate_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr4_M_ind_p,], file= "appleLate_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr4_H_ind_p,], file= "appleLate_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(Apple_late_mat[chr5_L_ind_p,], file= "appleLate_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr5_M_ind_p,], file= "appleLate_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Apple_late_mat[chr5_H_ind_p,], file= "appleLate_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

###Hawthorn early diapause termination 
#All chromosomes
write.table(Haw_early_mat[chr1_ind_p,], file= "HawEarly_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr2_ind_p,], file= "HawEarly_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr3_ind_p,], file= "HawEarly_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr4_ind_p,], file= "HawEarly_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr5_ind_p,], file= "HawEarly_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[all_chr_ind_p,], file= "HawEarly_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LD groups
write.table(Haw_early_mat[all_LDL_p,], file= "HawEarly_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[all_LDM_p,], file= "HawEarly_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[all_LDH_p,], file= "HawEarly_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(Haw_early_mat[chr1_L_ind_p,], file= "HawEarly_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr1_M_ind_p,], file= "HawEarly_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr1_H_ind_p,], file= "HawEarly_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(Haw_early_mat[chr2_L_ind_p,], file= "HawEarly_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr2_M_ind_p,], file= "HawEarly_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr2_H_ind_p,], file= "HawEarly_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(Haw_early_mat[chr3_L_ind_p,], file= "HawEarly_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr3_M_ind_p,], file= "HawEarly_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr3_H_ind_p,], file= "HawEarly_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(Haw_early_mat[chr4_L_ind_p,], file= "HawEarly_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr4_M_ind_p,], file= "HawEarly_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr4_H_ind_p,], file= "HawEarly_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(Haw_early_mat[chr5_L_ind_p,], file= "HawEarly_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr5_M_ind_p,], file= "HawEarly_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_early_mat[chr5_H_ind_p,], file= "HawEarly_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Hawthorn late termination
#All chromosomes
write.table(Haw_late_mat[chr1_ind_p,], file= "HawLate_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr2_ind_p,], file= "HawLate_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr3_ind_p,], file= "HawLate_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr4_ind_p,], file= "HawLate_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr5_ind_p,], file= "HawLate_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[all_chr_ind_p,], file= "HawLate_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All LD groups
write.table(Haw_late_mat[all_LDL_p,], file= "HawLate_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[all_LDM_p,], file= "HawLate_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[all_LDH_p,], file= "HawLate_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(Haw_late_mat[chr1_L_ind_p,], file= "HawLate_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr1_M_ind_p,], file= "HawLate_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr1_H_ind_p,], file= "HawLate_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(Haw_late_mat[chr2_L_ind_p,], file= "HawLate_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr2_M_ind_p,], file= "HawLate_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr2_H_ind_p,], file= "HawLate_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(Haw_late_mat[chr3_L_ind_p,], file= "HawLate_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr3_M_ind_p,], file= "HawLate_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr3_H_ind_p,], file= "HawLate_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(Haw_late_mat[chr4_L_ind_p,], file= "HawLate_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr4_M_ind_p,], file= "HawLate_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr4_H_ind_p,], file= "HawLate_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(Haw_late_mat[chr5_L_ind_p,], file= "HawLate_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr5_M_ind_p,], file= "HawLate_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(Haw_late_mat[chr5_H_ind_p,], file= "HawLate_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

setwd("../")

# Selection experimen, genotype data originally from Egan, Ragland et al. 2015 Ecol. Lett. 
Das <- read.vcfR("./data/snps.ScottSelection.05-400-20.vcf", nrows = -1, skip = 0, cols = NULL, convertNA = TRUE, verbose = T)
PLs <- extract.gt(Das, element = "PL", mask = F, as.numeric = F, return.alleles = F, IDtoRowNames = TRUE, extract = T)
r <- nrow(PLs)
c <- ncol(PLs)
mat <- matrix(lapply(PLs, splitPls), nrow=r, ncol=c)
rownames(mat)<-rownames(PLs)
colnames(mat)<-colnames(PLs)
ind <- rownames(mat) %in% jeff_snps #this is where Jeff's method falls apart a bit 2 loci arent on the list
mat <- mat[ind,]
rnames <- rownames(mat)
mat<- apply(mat,2,as.numeric)
row.names(mat) <- rnames
sel_scored <- matrix(nrow = nrow(mat), ncol = ncol(mat))
for(i in 1:length(pol_ind)){
  for(j in 1:ncol(sel_scored))
    if(pol_ind[i]==FALSE){
      sel_scored[i,j] <- mat[i,j]
    }else if(pol_ind[i]==TRUE){
      sel_scored[i,j] <- (2-mat[i,j])
    }
}
colnames(sel_scored) <- colnames(mat)
rownames(sel_scored) <- rownames(mat)

grant_ids <- read.table("./data/SelectionIdDescriptors.txt")
ind <- grep("Haw_[a-z]_7d", grant_ids$V2)
ind <- colnames(sel_scored) %in% grant_ids$V1[ind]
haw7_mat <- sel_scored[,ind]
ind <- grep("Haw_[a-z?]*_3[1-9]d", grant_ids$V2)
ind <- colnames(sel_scored) %in% grant_ids$V1[ind]
haw32_mat <- sel_scored[,ind]
ind <- grep("Apple_[a-z]_7d", grant_ids$V2)
ind <- colnames(sel_scored) %in% grant_ids$V1[ind]
apple7_mat <- sel_scored[,ind]
ind <- grep("Apple_[a-z]_32d", grant_ids$V2)
ind <- colnames(sel_scored) %in% grant_ids$V1[ind]
apple32_mat <- sel_scored[,ind]

setwd("./data/")

#Write the mean genotype files: 
#All chromosomes
write.table(apple7_mat[all_chr_ind_p,], file= "apple7_genos_AllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[all_chr_ind_p,], file= "apple32_genos_AllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[all_chr_ind_p,], file= "haw7_genos_AllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[all_chr_ind_p,], file= "haw32_genos_AllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All low LD
write.table(apple7_mat[all_LDL_p,], file= "apple7_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[all_LDL_p,], file= "apple32_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[all_LDL_p,], file= "haw7_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[all_LDL_p,], file= "haw32_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All int. LD
write.table(apple7_mat[all_LDM_p,], file= "apple7_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[all_LDM_p,], file= "apple32_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[all_LDM_p,], file= "haw7_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[all_LDM_p,], file= "haw32_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All high LD
write.table(apple7_mat[all_LDH_p,], file= "apple7_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[all_LDH_p,], file= "apple32_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[all_LDH_p,], file= "haw7_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[all_LDH_p,], file= "haw32_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All loci
write.table(apple7_mat, file= "apple7_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat, file= "apple32_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat, file= "haw7_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat, file= "haw32_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Apple - 7 day pre-winter treatment
#All chromosomes
write.table(apple7_mat[chr1_ind_p,], file= "apple7_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr2_ind_p,], file= "apple7_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr3_ind_p,], file= "apple7_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr4_ind_p,], file= "apple7_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr5_ind_p,], file= "apple7_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
#write.table(apple7_mat[all_chr_ind_p,], file= "apple7_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(apple7_mat[chr1_L_ind_p,], file= "apple7_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr1_M_ind_p,], file= "apple7_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr1_H_ind_p,], file= "apple7_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(apple7_mat[chr2_L_ind_p,], file= "apple7_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr2_M_ind_p,], file= "apple7_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr2_H_ind_p,], file= "apple7_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(apple7_mat[chr3_L_ind_p,], file= "apple7_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr3_M_ind_p,], file= "apple7_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr3_H_ind_p,], file= "apple7_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(apple7_mat[chr4_L_ind_p,], file= "apple7_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr4_M_ind_p,], file= "apple7_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr4_H_ind_p,], file= "apple7_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(apple7_mat[chr5_L_ind_p,], file= "apple7_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr5_M_ind_p,], file= "apple7_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple7_mat[chr5_H_ind_p,], file= "apple7_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Apple 32 day pre-winter treatment
#All chromosomes full
write.table(apple32_mat[chr1_ind_p,], file= "apple32_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr2_ind_p,], file= "apple32_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr3_ind_p,], file= "apple32_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr4_ind_p,], file= "apple32_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr5_ind_p,], file= "apple32_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
#write.table(apple32_mat[all_chr_ind_p,], file= "apple32_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(apple32_mat[chr1_L_ind_p,], file= "apple32_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr1_M_ind_p,], file= "apple32_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr1_H_ind_p,], file= "apple32_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(apple32_mat[chr2_L_ind_p,], file= "apple32_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr2_M_ind_p,], file= "apple32_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr2_H_ind_p,], file= "apple32_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(apple32_mat[chr3_L_ind_p,], file= "apple32_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr3_M_ind_p,], file= "apple32_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr3_H_ind_p,], file= "apple32_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(apple32_mat[chr4_L_ind_p,], file= "apple32_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr4_M_ind_p,], file= "apple32_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr4_H_ind_p,], file= "apple32_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(apple32_mat[chr5_L_ind_p,], file= "apple32_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr5_M_ind_p,], file= "apple32_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(apple32_mat[chr5_H_ind_p,], file= "apple32_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Hawthorn 7 day pre-winter treatment 
#All full chromosomes
write.table(haw7_mat[chr1_ind_p,], file= "haw7_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr2_ind_p,], file= "haw7_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr3_ind_p,], file= "haw7_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr4_ind_p,], file= "haw7_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr5_ind_p,], file= "haw7_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
#write.table(haw7_mat[all_chr_ind_p,], file= "haw7_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(haw7_mat[chr1_L_ind_p,], file= "haw7_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr1_M_ind_p,], file= "haw7_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr1_H_ind_p,], file= "haw7_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(haw7_mat[chr2_L_ind_p,], file= "haw7_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr2_M_ind_p,], file= "haw7_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr2_H_ind_p,], file= "haw7_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(haw7_mat[chr3_L_ind_p,], file= "haw7_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr3_M_ind_p,], file= "haw7_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr3_H_ind_p,], file= "haw7_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(haw7_mat[chr4_L_ind_p,], file= "haw7_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr4_M_ind_p,], file= "haw7_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr4_H_ind_p,], file= "haw7_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(haw7_mat[chr5_L_ind_p,], file= "haw7_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr5_M_ind_p,], file= "haw7_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw7_mat[chr5_H_ind_p,], file= "haw7_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

### Hawthorn - 32 day prewinter treatment
#All full chromosomes
write.table(haw32_mat[chr1_ind_p,], file= "haw32_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr2_ind_p,], file= "haw32_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr3_ind_p,], file= "haw32_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr4_ind_p,], file= "haw32_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr5_ind_p,], file= "haw32_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
#write.table(haw32_mat[all_chr_ind_p,], file= "haw32_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(haw32_mat[chr1_L_ind_p,], file= "haw32_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr1_M_ind_p,], file= "haw32_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr1_H_ind_p,], file= "haw32_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(haw32_mat[chr2_L_ind_p,], file= "haw32_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr2_M_ind_p,], file= "haw32_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr2_H_ind_p,], file= "haw32_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(haw32_mat[chr3_L_ind_p,], file= "haw32_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr3_M_ind_p,], file= "haw32_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr3_H_ind_p,], file= "haw32_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(haw32_mat[chr4_L_ind_p,], file= "haw32_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr4_M_ind_p,], file= "haw32_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr4_H_ind_p,], file= "haw32_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(haw32_mat[chr5_L_ind_p,], file= "haw32_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr5_M_ind_p,], file= "haw32_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(haw32_mat[chr5_H_ind_p,], file= "haw32_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

setwd("../")

# Diapause intensity data that was genotyped in this study 
#NOTE: The phenotype referred to as Diapause (DIA) in the manuscript is called CD in code here and elsewhere.
Das <- read.vcfR("./data/snps.GATK.all.filtered.vcf", nrows = -1, skip = 0, cols = NULL, convertNA = TRUE, verbose = T)
PLs <- extract.gt(Das, element = "PL", mask = F, as.numeric = F, return.alleles = F, IDtoRowNames = TRUE, extract = T)

r <- nrow(PLs)
c <- ncol(PLs)
mat <- matrix(lapply(PLs, splitPls), nrow=r, ncol=c)
rownames(mat)<-rownames(PLs)
colnames(mat)<-colnames(PLs)
ind <- rownames(mat) %in% jeff_snps
mat <- mat[ind,]
rnames <- rownames(mat)
mat<- apply(mat,2,as.numeric)
row.names(mat) <- rnames
dia_scored <- matrix(nrow = nrow(mat), ncol = ncol(mat))
for(i in 1:length(pol_ind)){
  for(j in 1:ncol(dia_scored))
    if(pol_ind[i]==FALSE){
      dia_scored[i,j] <- mat[i,j]
    }else if(pol_ind[i]==TRUE){
      dia_scored[i,j] <- (2-mat[i,j])
    }
}
colnames(dia_scored) <- colnames(mat)
rownames(dia_scored) <- rownames(mat)
r <- nrow(dia_scored)
c <- ncol(dia_scored)
SD_Ids <- read.table("./data/complete_shallowDiaIds.txt", header=F)
ND_Ids <- read.table("./data/complete_nonDiaIds.txt", header=F)
CD_Ids <- read.table("./data/complete_diaIds.txt", header=F)
SD_Ids <- as.vector(SD_Ids[,1])
SD_idx <- match(SD_Ids, colnames(mat))
SD_mat <- dia_scored[,SD_idx]
ND_Ids <- as.vector(ND_Ids[,1])
ND_idx <- match(ND_Ids, colnames(mat))
ND_mat <- dia_scored[,ND_idx]
CD_Ids <- as.vector(CD_Ids[,1])
CD_idx <- match(CD_Ids, colnames(mat))
CD_mat <- dia_scored[,CD_idx]
#Put CDs and NDs together because they are genetically very similar, (see the polygenic and BSLMM analyses) also
#we're mostly interested in the SD phenotype. 
CDND_mat <- as.matrix(cbind(CD_mat, ND_mat))

#Create the mean genotype data files 
setwd("./data/")


# Loci on chromosome 5 will be treated slightly differently. Unpublished work from Doellman, Feder, et al. has found strong evidence that the 
# sex determining region is located on chromosome 5. Given that the our sample sizes have a sex imbalance, we will use only the 
# females when estimating allele frequencies on chromosome 5. To make future analysis simpler, we will 
# therefore add NAs to each mean genotype for loci on chromosome 5 that belong to male samples. This is akin to remove the males from any analysis 
# involving chromosome 5. Downstream 

mer_Ids <- read.csv("RAD_Library_Info_PostSequencing_FixedForPlateSwitch_MDids.csv", header=T)
female_Ids <- read.table("ndsdcd.f.txt", header=F)
female_Ids <- as.character(female_Ids$V1)
SD_f_ids <- female_Ids[grep("sd", female_Ids)]
SD_f_ids <- as.character(mer_Ids$EcoRIAdapterName[mer_Ids$MD.ids %in% SD_f_ids])
CDND_f_ids <- female_Ids[grep("sd", female_Ids, invert = T)]
CDND_f_ids <- as.character(mer_Ids$EcoRIAdapterName[mer_Ids$MD.ids %in% DiaND_f_ids])
CD_f_ids <- female_Ids[grep("dia", female_Ids)]
CD_f_ids <- as.character(mer_Ids$EcoRIAdapterName[mer_Ids$MD.ids %in% CD_f_ids])
ND_f_ids <- female_Ids[grep("dia", female_Ids)]
ND_f_ids <- as.character(mer_Ids$EcoRIAdapterName[mer_Ids$MD.ids %in% CD_f_ids])

#All LD groups
#Have to slightly alter these LD indexes to capture just chromosomes 1-4, see chromosome assignments above.

write.table(SD_mat[chr1_4_LDL_p,], file= "SD_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_4_LDL_p,], file= "CDND_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(SD_mat[chr1_4_LDM_p,], file= "SD_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_4_LDM_p,], file= "CDND_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(SD_mat[chr1_4_LDH_p,], file= "SD_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_4_LDH_p,], file= "CDND_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#All genotyped loci

write.table(SD_mat, file= "SD_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat, file= "CDND_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

###Shallow diapause 
#All loci on chromosomes
write.table(SD_mat[chr1_ind_p,], file= "SD_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr2_ind_p,], file= "SD_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr3_ind_p,], file= "SD_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr4_ind_p,], file= "SD_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr5_ind_p, colnames(SD_mat) %in% SD_f_ids], file= "SD_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr1_4_all_P,], file= "SD_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(SD_mat[chr1_L_ind_p,], file= "SD_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr1_M_ind_p,], file= "SD_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr1_H_ind_p,], file= "SD_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(SD_mat[chr2_L_ind_p,], file= "SD_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr2_M_ind_p,], file= "SD_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr2_H_ind_p,], file= "SD_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(SD_mat[chr3_L_ind_p,], file= "SD_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr3_M_ind_p,], file= "SD_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr3_H_ind_p,], file= "SD_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(SD_mat[chr4_L_ind_p,], file= "SD_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr4_M_ind_p,], file= "SD_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr4_H_ind_p,], file= "SD_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD
write.table(SD_mat[chr5_L_ind_p, colnames(SD_mat) %in% SD_f_ids], file= "SD_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr5_M_ind_p, colnames(SD_mat) %in% SD_f_ids], file= "SD_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(SD_mat[chr5_H_ind_p, colnames(SD_mat) %in% SD_f_ids], file= "SD_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Combined Diapause and non diapause group. 

#All chromosomes
write.table(CDND_mat[chr1_ind_p,], file= "CDND_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr2_ind_p,], file= "CDND_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr3_ind_p,], file= "CDND_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr4_ind_p,], file= "CDND_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr5_ind_p,], file= "CDND_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_4_all_P,], file= "CDND_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 1 LD
write.table(CDND_mat[chr1_L_ind_p,], file= "CDND_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_M_ind_p,], file= "CDND_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr1_H_ind_p,], file= "CDND_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 2 LD
write.table(CDND_mat[chr2_L_ind_p,], file= "CDND_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr2_M_ind_p,], file= "CDND_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr2_H_ind_p,], file= "CDND_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 3 LD
write.table(CDND_mat[chr3_L_ind_p,], file= "CDND_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr3_M_ind_p,], file= "CDND_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr3_H_ind_p,], file= "CDND_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 4 LD
write.table(CDND_mat[chr4_L_ind_p,], file= "CDND_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr4_M_ind_p,], file= "CDND_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr4_H_ind_p,], file= "CDND_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Chromosome 5 LD 
# These loci will be treated slightly differently. Unpublished work from Doellman, Feder, et al. has found strong evidence that the 
# sex determining region is located on chromosome 5. Given that the our sample sizes have a sex imbalance, we will use only the 
# females from chromosome 5 

write.table(CDND_mat[chr5_L_ind_p, colnames(CDND_mat) %in% CDND_f_ids], file= "CDND_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr5_M_ind_p, colnames(CDND_mat) %in% CDND_f_ids], file= "CDND_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CDND_mat[chr5_H_ind_p, colnames(CDND_mat) %in% CDND_f_ids], file= "CDND_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#Dia and ND are below.

#Dia / CD

write.table(CD_mat[chr1_4_LDL_p,], file= "CD_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr1_4_LDM_p,], file= "CD_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr1_4_LDH_p,], file= "CD_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat, file= "CD_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")


write.table(CD_mat[chr1_ind_p,], file= "CD_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr2_ind_p,], file= "CD_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr3_ind_p,], file= "CD_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr4_ind_p,], file= "CD_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr5_ind_p, colnames(CD_mat) %in% CD_f_ids], file= "CD_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr1_4_all_P,], file= "CD_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(CD_mat[chr1_L_ind_p,], file= "CD_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr1_M_ind_p,], file= "CD_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(CD_mat[chr1_H_ind_p,], file= "CD_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

 write.table(CD_mat[chr2_L_ind_p,], file= "CD_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr2_M_ind_p,], file= "CD_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr2_H_ind_p,], file= "CD_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

 write.table(CD_mat[chr3_L_ind_p,], file= "CD_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr3_M_ind_p,], file= "CD_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr3_H_ind_p,], file= "CD_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

 write.table(CD_mat[chr4_L_ind_p,], file= "CD_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr4_M_ind_p,], file= "CD_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr4_H_ind_p,], file= "CD_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

 write.table(CD_mat[chr5_L_ind_p, colnames(CD_mat) %in% CD_f_ids], file= "CD_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr5_M_ind_p, colnames(CD_mat) %in% CD_f_ids], file= "CD_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
 write.table(CD_mat[chr5_H_ind_p, colnames(CD_mat) %in% CD_f_ids], file= "CD_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

#ND

write.table(ND_mat[chr1_4_LDL_p,], file= "ND_genos_AllLDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr1_4_LDM_p,], file= "ND_genos_AllLDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr1_4_LDH_p,], file= "ND_genos_AllLDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat, file= "ND_genosAll_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")


write.table(ND_mat[chr1_ind_p,], file= "ND_genosChr1_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr2_ind_p,], file= "ND_genosChr2_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr3_ind_p,], file= "ND_genosChr3_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr4_ind_p,], file= "ND_genosChr4_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr5_ind_p, colnames(CD_mat) %in% CD_f_ids], file= "ND_genosChr5_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[all_chr_ind_p,], file= "ND_genosAllChr_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(ND_mat[chr1_L_ind_p,], file= "ND_genosChr1_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr1_M_ind_p,], file= "ND_genosChr1_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr1_H_ind_p,], file= "ND_genosChr1_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(ND_mat[chr2_L_ind_p,], file= "ND_genosChr2_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr2_M_ind_p,], file= "ND_genosChr2_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr2_H_ind_p,], file= "ND_genosChr2_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(ND_mat[chr3_L_ind_p,], file= "ND_genosChr3_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr3_M_ind_p,], file= "ND_genosChr3_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr3_H_ind_p,], file= "ND_genosChr3_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(ND_mat[chr4_L_ind_p,], file= "ND_genosChr4_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr4_M_ind_p,], file= "ND_genosChr4_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr4_H_ind_p,], file= "ND_genosChr4_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

write.table(ND_mat[chr5_L_ind_p, colnames(ND_mat) %in% ND_f_ids], file= "ND_genosChr5_LDL_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr5_M_ind_p, colnames(ND_mat) %in% ND_f_ids], file= "ND_genosChr5_LDM_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")
write.table(ND_mat[chr5_H_ind_p, colnames(ND_mat) %in% ND_f_ids], file= "ND_genosChr5_LDH_polarized.txt", row.names = T, col.names = T, quote = F, sep = "\t")

